pkg/cloudprovider/external.go

```diff
@@ -17,12 +17,18 @@ const (
 // IsCloudProviderExternal is used to check whether external cloud provider settings should be used in a component.
 // It checks whether the ExternalCloudProvider feature gate is enabled and whether the ExternalCloudProvider feature
 // has been implemented for the platform.
-func IsCloudProviderExternal(platformType configv1.PlatformType, featureGate *configv1.FeatureGate) (bool, error) {
-	switch platformType {
+func IsCloudProviderExternal(infrastructureStatus configv1.InfrastructureStatus, featureGate *configv1.FeatureGate) (bool, error) {
+	switch infrastructureStatus.PlatformStatus.Type {
```

Comment:
Nothing checks whether `PlatformStatus` is non nil, we need to do this to avoid potential for panics

```suggestion
        if infrastructureStats.PlatformStatus == nil {
                return false, errors.New("platformStatus is required")
        }
	switch infrastructureStatus.PlatformStatus.Type {
```